docker/letsencrypt/letsencrypt-certbot/haproxy-reload.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh 
+
+# Update HAProxy with new certs via its API
+
+le_base=/etc/letsencrypt/live
+if [[ -d "$le_base" ]]; then
+    domains=$(find $le_base -mindepth 1 -type d -exec basename {} \;)
+    for domain in $domains; do
+        cert_path="/etc/haproxy/certs.d/$domain.pem"
+        # Start a transaction to update the certificate
+        echo -e "set ssl cert $cert_path <<\n$(cat $le_base/$domain/fullchain.pem $le_base/$domain/privkey.pem | sed '/^[[:blank:]]*$/ d')\n" | socat /var/lib/kolla/haproxy/haproxy.sock -
```

Comment:
It _might_ be a bit more readable to do that concat in a prior step

```bash
local cert = $(cat "$le_base/$domain/fullchain.pem" "$le_base/$domain/privkey.pem" | sed '/^[[:blank:]]*$/ d')
echo -e "set ssl cert $cert_path $cert\n" | socat /var/lib/kolla/haproxy/haproxy.sock -
```

though I admit there are 🐉 🐉 in shell quoting and perhaps there is some edge case there.

Reply:
Copied almost verbatim from [this post](https://www.haproxy.com/blog/dynamic-ssl-certificate-storage-in-haproxy/) 🤷

I agree that splitting it up is more readable. The amount of time it would take to test such a small change is what upsets me 😭

Reply:
agreed on the two line part, if only because I'm going to be baffled when I come across that line a year from now.

Is "set ssl cert" the part that begins the transaction mentioned in the comment?

Reply:
> Is "set ssl cert" the part that begins the transaction mentioned in the comment?

Correct.